core/executor/src/host_interface.rs

```diff
@@ -790,7 +790,39 @@ impl_wasm_host_interface! {
 
 			context.write_memory(pubkey_data, &pubkey.serialize()[1..65])
 				.map_err(|_| "Invalid attempt to set pubkey in ext_secp256k1_ecdsa_recover")?;
+			Ok(0)
+		}
+
+		ext_secp256k1_ecdsa_recover_compressed(
+			msg_data: Pointer<u8>,
+			sig_data: Pointer<u8>,
+			pubkey_data: Pointer<u8>,
+		) -> u32 {
+			let mut sig = [0u8; 65];
```

Comment:
Could you at least factor out the common code between this and `ext_secp256k1_ecdsa_recover` into a separate function?